web/src/engine/websites/MangaUpGlobal.ts

```diff
@@ -51,29 +62,51 @@ export default class extends DecoratableMangaScraper {
-        const data = await this.FetchNextData<APIMangas>(new URL('/search', this.URI));
-        return data.titles.map(manga => new Manga(this, provider, manga.titleId, manga.titleName.trim()));
+        const request = new Request(new URL(`search`, this.apiUrl));
+        const { titles } = await FetchProto<APISearch>(request, protoTypes, 'MangaUpGlobal.SearchView');
+        return titles.map(manga => new Manga(this, provider, manga.titleId.toString(), manga.titleName.trim()));
     }
 
     public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const data = await this.FetchNextData<APIManga>(new URL(`/manga/${manga.Identifier}`, this.URI));
-        return data.chapters.map(chapter => new Chapter(this, manga, chapter.id.toString(), chapter.mainName.trim()));
+        const request = new Request(new URL(`manga/detail_v2?title_id=${manga.Identifier}`, this.apiUrl));
+        const { chapters } = await FetchProto<APIMangaDetailView>(request, protoTypes, 'MangaUpGlobal.MangaDetailView');
+        return chapters.map(chapter => new Chapter(this, manga, chapter.id.toString(), [chapter.titleName, chapter.subName].join(' ').trim()));
     }
 
-    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
-        const data = await this.FetchNextData<APIPages>(new URL(`/manga/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.URI));
-        if (!data)
-            throw new Exception(W.Plugin_Common_Chapter_UnavailableError);
-        return data.pages.map(image => new Page(this, chapter, new URL(image.imageUrl)));
+    public override async FetchPages(chapter: Chapter): Promise<Page<CryptoParams>[]> {
+        const request = new Request(new URL(`manga/viewer_v2?chapter_id=${chapter.Identifier}&quality=high`, this.apiUrl), { method: 'POST' });
+        const data = await FetchProto<APIPages>(request, protoTypes, 'MangaUpGlobal.MangaViewerV2View');
+
+        if (!data.pageblocks) throw new Exception(W.Plugin_Common_Chapter_UnavailableError);
+
+        return data.pageblocks.shift().pages.map(page => {
+            const params: CryptoParams = {
+                method: page.iv ? 'aes-cbc' : undefined,
+                key: page.encryptionKey,
+                iv: page.iv
+            };
+            return new Page<CryptoParams>(this, chapter, new URL(page.imageUrl, this.imagesCDN), params);
+        });
     }
 
-    private async FetchNextData<T extends JSONElement>(url: URL): Promise<T> {
-        const data = await FetchWindowScript<NEXTDATA<T>>(new Request(url), '__NEXT_DATA__', 2000);
-        return data.props.pageProps.data as T;
+    public override async FetchImage(page: Page<CryptoParams>, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal, true);
+        const cryptoParams = page.Parameters;
+        switch (cryptoParams.method) {
```

Comment:
Yeah, Its like that, because website have code that is perhaps used on another website where iv is extracted from buffer, and xor method.
Afaik this is unused in MangaUp